Formula/flutter.rb

```diff
@@ -0,0 +1,27 @@
+class Flutter < Formula
+  desc "Google’s UI toolkit for mobile, web, and desktop from a single codebase"
+  homepage "https://flutter.dev"
+  url "https://storage.googleapis.com/flutter_infra/releases/stable/macos/flutter_macos_v1.9.1+hotfix.6-stable.zip"
+  sha256 "8d0b3e217e45fbde64e117c5932ec5bd18ced0e8e8fba80a0fec95e38854bb6a"
+
+  def install
+    cp_r ".", prefix
+  end
+
+  def post_install
+    chmod_R "u+w", prefix/"bin/cache"
+  end
+
+  def caveats; <<~EOS
+    Run the following command to see if there are any dependencies you need to install to complete the setup (for verbose output, add the -v flag):
+      $ flutter doctor
+    This command checks your environment and displays a report to the terminal window.
+    The Dart SDK is bundled with Flutter; it is not necessary to install Dart separately.
+    Check the output carefully for other software you might need to install or further tasks to perform (shown in bold text).
+  EOS
+  end
+
+  test do
+    assert_predicate bin/"flutter", :exist?
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.